src/components/user/user.service.ts

```diff
@@ -227,22 +229,39 @@ export class UserService {
     // ensure token doesn't have any users attached to it
     // await this.logout(session.token);
 
+    // helper method for defining properties
+    const property = (prop: string, value: any) => {
+      if (!value) {
+        return [];
+      }
+
+      return [
+        node('user'),
+        relation('out', '', prop, {
+          active: true,
+          createdAt,
```

Comment:
This can reference the variable defined below?

Reply:
const vars get hoisted. But I can see how that looks a little weird, I'll move it to the top

Reply:
Okay! latest updates pushed!

Reply:
Yeah you're right. But I agree it looks better now, thanks!